docs/slotfilling.rst

```diff
@@ -3,12 +3,284 @@
 Slot Filling
 ============
 
-One of the most common conversation patterns
-is to collect a few pieces of information
-from a user in order to do something (book a restaurant, call an API, search a database, etc.).
-This is also called **slot filling**.
+One of the most common conversation patterns is to collect a few pieces of
+information from a user in order to do something (book a restaurant, call an
+API, search a database, etc.). This is also called **slot filling**. Our
+recommended approach for this is the FormAction, which is described in detail
+below.
 
+Slot Filling with a ``FormAction``
+----------------------------------
+
+If you need to collect multiple pieces of information in a row, it is recommended
+to create a ``FormAction``.
+This lets you have a single action that is called multiple times, rather than
+separate actions for each question. The idea behind this is that you can get the
+happy path of your bot up and running quickly. If we take the example of the
+restaurant bot, this means your initial story will look as simple as this
+(provided your slots are `unfeaturized
+<https://rasa.com/docs/core/api/slots_api/#unfeaturized-slot>`_):
+
+.. code-block:: story
+
+    ## happy path
+    * request_restaurant
+        - restaurant_form
+        - form{"name": "restaurant_form"}
+        - form{"name": null}
+
+
+The ``restaurant_form`` is the form action, for which you need to define a few
+methods in order for it to work. The basic class you need to subclass looks
+like this:
+
+.. code-block:: python
```

Comment:
it's not the whole base class, it's only the relevant methods you should override. And we go through how to implement each method in a real action below, which i think makes more sense since you want each method in front of you when talking about it

Reply:
yes but it's a little intimidating to dump the whole class here right at the start. I think [automethod](http://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#directive-automethod) might be our friend here

Reply:
ok i'll just do that then, but leave the definition of each method in their individual sections?